store/models/run.go

```diff
@@ -172,11 +184,13 @@ func (tr TaskRun) MarkPendingConfirmations() TaskRun {
 // RunResult keeps track of the outcome of a TaskRun or JobRun. It stores the
 // Data and ErrorMessage, and contains a field to track the status.
 type RunResult struct {
```

Comment:
Seems like this should be polymorphic. [Made a story](https://www.pivotaltracker.com/story/show/163375008) to tackle in the future.

Reply:
I think otherwise, due to the inherit complexity of polymorphism and the inability to do foreign key constraints, we'll save the chat for standup/IPM. Regardless, as you said, for a future PR.